<!DOCTYPE html>
<html lang="en"><head><meta charset="UTF-8"/><meta name="viewport" content="width=device-width, initial-scale=1.0"/><title>Home · GeCKOReader.jl</title><link href="https://cdnjs.cloudflare.com/ajax/libs/normalize/4.2.0/normalize.min.css" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.5.0/styles/default.min.css" rel="stylesheet" type="text/css"/><link href="https://fonts.googleapis.com/css?family=Lato|Ubuntu+Mono" rel="stylesheet" type="text/css"/><link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" type="text/css"/><link href="assets/documenter.css" rel="stylesheet" type="text/css"/><script>documenterBaseURL="."</script><script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.2.0/require.min.js" data-main="assets/documenter.js"></script><script src="siteinfo.js"></script><script src="../versions.js"></script></head><body><nav class="toc"><h1>GeCKOReader.jl</h1><select id="version-selector" onChange="window.location.href=this.value" style="visibility: hidden"></select><form class="search" action="search.html"><input id="search-query" name="q" type="text" placeholder="Search docs"/></form><ul><li class="current"><a class="toctext" href="index.html">Home</a><ul class="internal"><li><a class="toctext" href="#Loading-1">Loading</a></li><li><a class="toctext" href="#Processing-1">Processing</a></li><li><a class="toctext" href="#Plotting-1">Plotting</a></li></ul></li></ul></nav><article id="docs"><header><nav><ul><li><a href="index.html">Home</a></li></ul><a class="edit-page" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/docs/src/index.md"><span class="fa"></span> Edit on GitHub</a></nav><hr/></header><h1><a class="nav-anchor" id="GeCKOReader.jl-1" href="#GeCKOReader.jl-1">GeCKOReader.jl</a></h1><p><img src="https://img.shields.io/badge/status-wip-red.svg" alt="Project Status"/><a href="https://travis-ci.org/tlnagy/GeCKOReader.jl"><img src="https://travis-ci.org/tlnagy/GeCKOReader.jl.svg?branch=master" alt="Build Status"/></a></p><p>Some analysis code for interpreting GeCKOv2 CRISPR knockout screens</p><h2><a class="nav-anchor" id="Loading-1" href="#Loading-1">Loading</a></h2><p>These functions are useful for loading <code>fastq.gz</code> files</p><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.read_seq_file-Tuple{Dict{Bio.Seq.BioSequence{Bio.Seq.DNAAlphabet{4}},String},String,Bio.Seq.BioSequence{Bio.Seq.DNAAlphabet{4}},Int64,Int64,Int64}" href="#GeCKOReader.read_seq_file-Tuple{Dict{Bio.Seq.BioSequence{Bio.Seq.DNAAlphabet{4}},String},String,Bio.Seq.BioSequence{Bio.Seq.DNAAlphabet{4}},Int64,Int64,Int64}"><code>GeCKOReader.read_seq_file</code></a> — <span class="docstring-category">Method</span>.</div><div><p>read_seq_file(filename, barcode, stagger, read_length; verbose=false)</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/io.jl#L1-L4">source</a><br/></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.gen_barcode_mapping-Tuple{DataFrames.DataFrame,Int64}" href="#GeCKOReader.gen_barcode_mapping-Tuple{DataFrames.DataFrame,Int64}"><code>GeCKOReader.gen_barcode_mapping</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">gen_barcode_mapping(barcodes::DataFrame, len::Int)</code></pre><p>Generate a dictionary mapping of sequence of type Bio.Seq.DNASequence to their string id from <code>barcodes</code>. Truncate all barcodes to length <code>len</code>. The final result is not guaranteed to be unique due to the truncation step.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/utils.jl#L19-L25">source</a><br/></section><h2><a class="nav-anchor" id="Processing-1" href="#Processing-1">Processing</a></h2><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.calc_log2fc-Tuple{DataFrames.DataFrame,DataFrames.DataFrame}" href="#GeCKOReader.calc_log2fc-Tuple{DataFrames.DataFrame,DataFrames.DataFrame}"><code>GeCKOReader.calc_log2fc</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">calc_log2fc(exp::DataFrame, control::DataFrame)</code></pre><p>Combine <code>control</code> with <code>exp</code> and compute the log2 fold change between their <code>:rel_freqs</code> columns</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/processing.jl#L69-L74">source</a><br/></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.compute_gene_stats-Tuple{DataFrames.DataFrame,Array{Float64,1}}" href="#GeCKOReader.compute_gene_stats-Tuple{DataFrames.DataFrame,Array{Float64,1}}"><code>GeCKOReader.compute_gene_stats</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">compute_gene_stats(df::DataFrame, negcontrols_fc::Vector{Float64};
                   column::Symbol=:log2fc)</code></pre><p>Calculates pvalues and medians on a per gene basis for the given DataFrame <code>df</code> using the specified <code>column</code>. P-values are computed using a Mann-Whitney test of the gene values against the values provided in <code>negcontrols_fc</code> for the negative controls.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/processing.jl#L91-L99">source</a><br/></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.fit_lines-Tuple{DataFrames.DataFrame,DataFrames.DataFrame,DataFrames.DataFrame}" href="#GeCKOReader.fit_lines-Tuple{DataFrames.DataFrame,DataFrames.DataFrame,DataFrames.DataFrame}"><code>GeCKOReader.fit_lines</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">fit_lines(t0::DataFrame, t1::DataFrame, t2::DataFrame)</code></pre><p>Fit a line to the relative frequencies of each guide at <code>t0</code>, <code>t1</code>, and <code>t2</code> by solving the linear least squares equation for an overdetermined system.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/processing.jl#L122-L127">source</a><br/></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.get_log2fc_pvals-Tuple{DataFrames.DataFrame,DataFrames.DataFrame}" href="#GeCKOReader.get_log2fc_pvals-Tuple{DataFrames.DataFrame,DataFrames.DataFrame}"><code>GeCKOReader.get_log2fc_pvals</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">get_log2fc_pvals(exp::DataFrame, control::DataFrame)</code></pre><p>Given a treatment dataframe <code>exp</code> and a control dataframe <code>control</code> calculates the log 2 fold changes on a per guide level and then uses a Mann Whitney test to compare this the negative control guides to determine the significance on a per gene level.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/processing.jl#L40-L47">source</a><br/></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.get_rel_freqs-Tuple{GeCKOReader.ScreenSample,DataFrames.DataFrame}" href="#GeCKOReader.get_rel_freqs-Tuple{GeCKOReader.ScreenSample,DataFrames.DataFrame}"><code>GeCKOReader.get_rel_freqs</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">get_rel_freqs(sample::ScreenSample,
                   callable_bc::DataFrame)::DataFrame</code></pre><p>Computes the relative frequencies of barcodes as compared to the median frequency of occurence for all the negative controls. The frequencies are computed initially by dividing by the total number of reads for that sample.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/processing.jl#L1-L8">source</a><br/></section><h2><a class="nav-anchor" id="Plotting-1" href="#Plotting-1">Plotting</a></h2><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.Plot.plot_all_qualities-Tuple{GeCKOReader.ScreenSample}" href="#GeCKOReader.Plot.plot_all_qualities-Tuple{GeCKOReader.ScreenSample}"><code>GeCKOReader.Plot.plot_all_qualities</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">plot_all_qualities(s::ScreenSample)</code></pre><p>Generate a 2D histogram of read qualities from the FASTQ file per position in the read using <strong>all</strong> of the read info.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/plotting.jl#L27-L32">source</a><br/></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.Plot.plot_guide_trajectories-Tuple{DataFrames.DataFrame,String}" href="#GeCKOReader.Plot.plot_guide_trajectories-Tuple{DataFrames.DataFrame,String}"><code>GeCKOReader.Plot.plot_guide_trajectories</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">plot_guide_trajectories(lines::DataFrame, gene::String)</code></pre><p>Plot the trajectories of the guides for a specific <code>gene</code> given the data in <code>lines</code>. The <code>lines</code> dataframe should be formatted like the output of <a href="index.html#GeCKOReader.fit_lines-Tuple{DataFrames.DataFrame,DataFrames.DataFrame,DataFrames.DataFrame}"><code>fit_lines</code></a>.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/plotting.jl#L45-L51">source</a><br/></section><section class="docstring"><div class="docstring-header"><a class="docstring-binding" id="GeCKOReader.Plot.plot_hit_qualities-Tuple{GeCKOReader.ScreenSample}" href="#GeCKOReader.Plot.plot_hit_qualities-Tuple{GeCKOReader.ScreenSample}"><code>GeCKOReader.Plot.plot_hit_qualities</code></a> — <span class="docstring-category">Method</span>.</div><div><pre><code class="language-none">plot_hit_qualities(s::ScreenSample)</code></pre><p>Generate a 2D histogram of read qualities from the FASTQ file per position in the read using <strong>only reads that were matched</strong>.</p></div><a class="source-link" target="_blank" href="https://github.com/tlnagy/GeCKOReader.jl/tree/a58fa07c9c9c362e6aed5b6b5e7555ea6ba68b57/src/plotting.jl#L36-L41">source</a><br/></section><footer><hr/></footer></article></body></html>
